### PR TITLE
feat: parse console scripts

### PR DESCRIPTION
**Pull request by nichmor**
## Overview 

This fix/feature is necessary for windows pipeline failure : https://github.com/prefix-dev/pixi/pull/2638

I've added some unit tests for parsing console scripts. Integration test itself will be this PR: https://github.com/prefix-dev/pixi/pull/2638

Before merging this need to land:
https://github.com/prefix-dev/pixi/pull/2643

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -37,10 +37,15 @@ rattler_conda_types = { version = "0.29.1", default-features = false }
...
 rattler_virtual_packages = { version = "1.1.10", default-features = false }
 
-pixi_build_types = { git = "https://github.com/prefix-dev/pixi", branch = "main" }
-pixi_consts = { git = "https://github.com/prefix-dev/pixi", branch = "main" }
-pixi_manifest = { git = "https://github.com/prefix-dev/pixi", branch = "main" }
-pixi_spec = { git = "https://github.com/prefix-dev/pixi", branch = "main" }
+# pixi_build_types = { git = "https://github.com/prefix-dev/pixi", branch = "main" }
+# pixi_consts = { git = "https://github.com/prefix-dev/pixi", branch = "main" }
+# pixi_manifest = { git = "https://github.com/prefix-dev/pixi", branch = "main" }
+pixi_manifest = { path = "/Users/graf/projects/oss/pixi-build-branch/crates/pixi_manifest" }
```
Comment (wolfv) [inline Cargo.toml:43]: pls point to git.

File: crates/pixi-build/src/bin/pixi-build-python/python.rs [hunk 2]
```diff
@@ -204,6 +206,19 @@ impl PythonBuildBackend {
             NoArchType::none()
         };
 
+        // Determine the entry points from the pyproject.toml
+        // which would be passed into recipe
+        let python = if self.manifest.document.is_pyproject_toml() {
```
Comment (wolfv) [inline crates/pixi-build/src/bin/pixi-build-python/python.rs:211]: Hmm, I wonder if we should just re-parse the file as pyproject.toml?

File: Cargo.toml (hunk 1 above)
Comment (nichmor) [reply Cargo.toml]: changed to main as the PR that blocked me was merged

**Review by wolfv — APPROVED** [review]

File: crates/pixi-build/src/bin/pixi-build-python/python.rs (hunk 2 above)
Comment (nichmor) [reply crates/pixi-build/src/bin/pixi-build-python/python.rs:211]: ah got it now! you meant with the pyproject-toml library

**Review by Hofer-Julian — APPROVED** [review]
One tiny suggestion. Apart from that it looks great @nichmor!

**Pull request merged**